plugins/woocommerce/src/Caches/OrderAggregateCache.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Caches;
+
+use Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore;
+
+/**
+ * A class to cache aggregates from orders.
+ */
+class OrderAggregateCache {
+
+	/**
+	 * Option name.
+	 */
+	const OPTION_NAME = 'woocommerce_orders_aggregate_cache';
+
+	/**
+	 * The running totals for all order statuses.
+	 *
+	 * @var array|null
+	 */
+	private $totals = null;
+
+	/**
+	 * Whether or not the count has changed since loading.
+	 *
+	 * @var bool
+	 */
+	private $has_changed = false;
+
+	/**
+	 * Get the aggregate by identifier.
+	 *
+	 * @param string|string[] $status The order status to count.
+	 * @return int
+	 * @throws \Exception Exception thrown if status it not valid.
+	 */
+	public function get_count( $status ): int {
+		$status         = (array) $status;
+		$valid_statuses = $this->get_valid_statuses();
+
+		if ( ! empty( array_diff( $status, $valid_statuses ) ) ) {
+			throw new \Exception();
```

Comment:
A bit too generic =)

Reply:
Just a placeholder until we decide to move forward with this concept 😄

Reply:
These now use `CacheException`.